src/core/activityContent/validation.ts

```diff
@@ -40,386 +42,398 @@ const SUPPORTED_IMAGE_MEDIA_TYPES = [
+  requireIsMediaContentLink(obj, "ActivityContentAudioLink");
+  return true;
+};
 
-  if (obj["name"] && !isString(obj["name"])) return false;
+const requireIsActivityContentImageLink = (obj: unknown): obj is ActivityContentImageLink => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(invalidRecordMessage("ActivityContentImageLink"));
+  requireIsMediaContentLink(obj, "ActivityContentImageLink");
+  requireValidDimensions(obj, "ActivityContentImageLink");
+  return true;
+};
 
+const requireIsActivityContentVideoLink = (obj: unknown): obj is ActivityContentVideoLink => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(invalidRecordMessage("ActivityContentVideoLink"));
+  requireIsMediaContentLink(obj, "ActivityContentVideoLink");
+  requireValidDimensions(obj, "ActivityContentVideoLink");
   return true;
 };
 
-const isActivityContentAudio = (obj: unknown): obj is ActivityContentAudio => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Audio") return false;
-  if (!isArrayOfType(obj["url"], isActivityContentAudioLink)) return false;
+const requireToActivityContentLink = (obj: unknown): ActivityContentLink => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(invalidRecordMessage("ActivityContentVideoLink"));
+  if (obj["type"] !== "Link") throw new InvalidActivityContentError("ActivityContentLink type is not 'Link'");
 
-  if (obj["name"] && !isString(obj["name"])) return false;
-  if (obj["duration"] && !isString(obj["duration"])) return false;
+  const href = requireToString(obj["href"], "ActivityContentLink href");
+  if (!isValidHrefField(href)) throw new InvalidActivityContentError("ActivityContentLink href is invalid");
 
-  return true;
-};
+  if (obj["name"]) requireToString(obj["name"], "ActivityContentLink name");
 
-const isActivityContentImage = (obj: unknown): obj is ActivityContentImage => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Image") return false;
-  if (!isArrayOfType(obj["url"], isActivityContentImageLink)) return false;
+  return { type: "Link", href: "", ...obj };
+};
 
-  if (obj["name"] && !isString(obj["name"])) return false;
+const requireIsActivityContentHash = (obj: unknown): obj is ActivityContentHash => {
+  if (!isRecord(obj)) throw new InvalidActivityContentError(invalidRecordMessage("ActivityContentHash"));
+  const hashVal = requireToString(obj["value"], "ActivityContentHash value field");
+  if (!hashVal.match(HASH_REGEX)) throw new InvalidActivityContentError("ActivityContentHash value is invalid");
 
+  requireToString(obj["algorithm"], "ActivityContentHash algorithm");
   return true;
 };
 
-const isActivityContentVideo = (obj: unknown): obj is ActivityContentVideo => {
-  if (!isRecord(obj)) return false;
-  if (obj["type"] !== "Video") return false;
-  if (!isArrayOfType(obj["url"], isActivityContentVideoLink)) return false;
+const requireIsActivityContentLocation = (obj: unknown): obj is ActivityContentLocation => {
```

Comment:
This validation appears to have loosened quite a bit (i.e. no restrictions on units or bounds on values). Is that ok? The previous version seems dangerously strict to me.

Reply:
There's a lot of potential range on location data, and both decimal and negative values are valid. I think those kinds of checks should be left to consumers of that optional data.